homeassistant/components/binary_sensor/bmw_connected_drive.py

```diff
@@ -92,12 +110,39 @@ def device_state_attributes(self):
                 result[window.name] = window.state.value
         elif self._attribute == 'door_lock_state':
             result['door_lock_state'] = vehicle_state.door_lock_state.value
+            result['last_update_reason'] = vehicle_state.last_update_reason
+        elif self._attribute == 'lights_parking':
+            result['lights_parking'] = vehicle_state.parking_lights.value
+        elif self._attribute == 'condition_based_services':
+            for report in vehicle_state.condition_based_services:
+                service_type = report.service_type.lower().replace('_', ' ')
+                result['{} status'.format(service_type)] = report.state.value
+                if report.due_date is not None:
+                    result['{} date'.format(service_type)] = \
+                        report.due_date.strftime('%Y-%m-%d')
+                if report.due_distance is not None:
+                    result['{} distance'.format(service_type)] = \
+                        '{} km'.format(report.due_distance)
+        elif self._attribute == 'check_control_messages':
+            check_control_messages = vehicle_state.check_control_messages
+            if not check_control_messages:
+                result['check_control_messages'] = 'OK'
+            else:
+                result['check_control_messages'] = check_control_messages
+        elif self._attribute == 'charging_status':
+            result['charging_status'] = vehicle_state.charging_status.value
+            result['last_charging_end_result'] = \
+                vehicle_state._attributes['lastChargingEndResult']
```

Comment:
Travis is failing on three of these accesses to private attributes. You can disable that check above those lines. But you should probably change this in the library so that we don't have to access private attributes.

Reply:
Thanks! Have disabled those checks for now. Will put it on the to do list to change this in the library.